app/src/utils/spatial-utils.tsx

```diff
@@ -0,0 +1,194 @@
+import { IMarker, IMarkerLayer } from 'components/map/components/MarkerCluster';
+import { IStaticLayer } from 'components/map/components/StaticLayers';
+import DatasetPopup from 'components/map/DatasetPopup';
+import FeaturePopup, { BoundaryCentroidFeature, BoundaryFeature, OccurrenceFeature } from 'components/map/FeaturePopup';
+import { LAYER_NAME, SPATIAL_COMPONENT_TYPE } from 'constants/spatial';
+import { Feature } from 'geojson';
+import { EmptyObject, ISpatialData, ITaxaData } from 'interfaces/useObservationApi.interface';
+import { LatLngTuple } from 'leaflet';
+import { isObject } from 'lodash-es';
+import React from 'react';
+
+export interface IDataResult {
+  key: string;
+  name: string;
+  count: number;
+}
+
+export interface ISpatialDataGroupedBySpecies {
+  [species: string]: ISpatialData[];
+}
+
+export const parseSpatialDataByType = (spatialDataRecords: ISpatialData[]) => {
+  const occurrencesMarkerLayer: IMarkerLayer = { layerName: LAYER_NAME.OCCURRENCES, markers: [] };
+  const boundaryStaticLayer: IStaticLayer = { layerName: LAYER_NAME.BOUNDARIES, features: [] };
+
+  for (const spatialRecord of spatialDataRecords) {
+    for (const feature of spatialRecord.spatial_data.features) {
+      const visible = true;
+
+      if (feature.geometry.type === 'GeometryCollection') {
+        // Not expecting or supporting geometry collections
+        continue;
+      }
+
+      if (isOccurrenceFeature(feature)) {
+        // check if species has been toggled on/ off
+        const marker = occurrenceMarkerSetup(feature.geometry.coordinates as LatLngTuple, spatialRecord.taxa_data);
+        if (marker) {
+          occurrencesMarkerLayer.markers.push(marker);
+        }
+      }
+
+      if (isBoundaryFeature(feature)) {
+        // check if dataset has been toggled
+        const ids = getSubmissionSpatialComponentIds(spatialRecord);
+
+        if (visible) {
```

Comment:
Do we expect this check to ever be `false`? `visible` is a const so we can't redefine it. Should we remove these checks?